addons/project/tests/test_project_sharing_portal_access.py

```diff
@@ -76,3 +79,40 @@ def test_write_with_portal_user(self):
         for field in self.other_fields_task:
             with self.assertRaises(AccessError):
                 self.task_portal.with_user(self.user_portal).write({field: 'dummy'})
+
+
+class TestProjectSharingChatterAccess(TestProjectSharingCommon, HttpCase):
+
+    def test_post_chatter_as_portal_user(self):
+        self.env['project.share.wizard'].create({
+            'res_model': 'project.project',
+            'res_id': self.project_no_collabo.id,
+            'access_mode': 'edit',
+            'partner_ids': [Command.set([self.user_portal.partner_id.id])],
+        }).action_send_mail()
+        message = self.env['mail.message'].search([
+            ('partner_ids', 'in', self.user_portal.partner_id.id),
+        ])
+        share_link = str(message.body.split('href="')[1].split('">')[0])
+        access_token = share_link.split('&amp;')[2][13:]
+        pid = share_link.split('&amp;')[3][4:]
+        _hash = share_link.split('&amp;')[4][5:]
```

Comment:
Actually, cannot we do a regex to correctly select the right value (to have something more robust, in the case the order changed).